src/core/core.datasetController.js

```diff
@@ -234,6 +235,17 @@ helpers.extend(DatasetController.prototype, {
 		me.resyncElements();
 	},
 
+	/**
+	 * Returns the default dataset-level options
+	 * @private
+	 */
+	_defaults: function() {
+		var me = this;
+		var dataset = me.getDataset();
+		var type = helpers.valueOrDefault(dataset.type, me.chart.config.type);
```

Comment:
We shouldn't "resolve" the dataset type because (if I'm not wrong) [it's already part of the dataset meta](/chartjs/Chart.js/blob/master/src/core/core.controller.js#L342). Since the type is constant and querying the dataset meta is not optimal, I would store the type as a private member in `initialize()` (after `me.index = ...`).

```
me.index = datasetIndex;
me._type = me.getMeta().type;
```

Then `_defaults()` becomes:
```javascript
return defaults.datasets[this._type] || {};
```

I would return an empty object in case there is no dataset defaults.

Reply:
Done